modular_skyrat/modules/customization/modules/clothing/storage/belts.dm

```diff
@@ -0,0 +1,115 @@
+		to_chat(user, "<span class='warning'>[src] is empty!</span>")
+	. = ..()
+
+/obj/item/storage/belt/crusader/update_appearance(updates)
+	if(contents.len == 2)	//Checks for a sword/rod in the sheath slot, changes the sprite accordingly
+		icon_state = "crusader_belt_sheathed"
+		worn_icon_state = "crusader_belt_sheathed"
+	else
+		icon_state = "crusader_belt"
+		worn_icon_state = "crusader_belt"
+	. = ..()
+
+/obj/item/storage/belt/crusader/examine(mob/user)
+	. = ..()
+	.+= "<span class='notice'>Ctrl-click it to easily open its inventory.</span>"
+	if(contents.len == 2)	//If there's no sword/rod in the sheath slot it doesnt display the alt-click instruction
+		. += "<span class='notice'>Alt-click it to quickly draw the blade.</span>"
+		return
+
+/obj/item/storage/belt/crusader/ComponentInitialize()
+	. = ..()
+	AddElement(/datum/element/update_icon_updates_onmob)
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+
+	STR.max_items = 2
+	STR.rustle_sound = TRUE
+	STR.max_w_class = WEIGHT_CLASS_GIGANTIC	//Max size makes sure that the storage_pouch item still properly fits in it - the whitelist keeps anything fucky out
+	STR.allow_big_nesting = TRUE //Same as above
+	STR.set_holdable(list(
+		/obj/item/storage/belt/storage_pouch,
+		/obj/item/melee/sabre,
+		/obj/item/melee/cleric_mace,
+		/obj/item/nullrod	//holds any subset of nullrod in the sheath-storage, so this doesnt become a massive list
+		))
+
+/obj/item/storage/belt/crusader/PopulateContents()
+	. = ..()
+	new /obj/item/storage/belt/storage_pouch(src)
+	return
+
+/obj/item/storage/belt/storage_pouch	//seperate mini-storage inside the belt, leaving room for only one sword. Inspired by a (very poorly implemented) belt on Desert Rose
+	icon = 'modular_skyrat/modules/customization/icons/obj/clothing/storage.dmi'
+	worn_icon = 'modular_skyrat/modules/customization/icons/mob/clothing/storage.dmi'
+	name = "storage pouch"
+	desc = "<span class='notice'>Click on this to open your belt's inventory!</span>"
+	icon_state = "storage_pouch_icon"
+	worn_icon_state = "no name"	//Intentionally sets the worn icon to an error
+	w_class = WEIGHT_CLASS_GIGANTIC	//Keeps it from being dragged into other bags
+	anchored = 1	//Dont want people taking it out with their hands
+
+/obj/item/storage/belt/storage_pouch/attack_hand(mob/user, list/modifiers)	//Opens the bag on click - considering it's already anchored, this makes it function similar to how ghosts can open all nested inventories
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.user_show_to_mob(user)
+	return
+
+/obj/item/storage/belt/storage_pouch/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+
+	STR.max_items = 6
+	STR.rustle_sound = TRUE
+	STR.max_w_class = WEIGHT_CLASS_SMALL //Rather than have a huge whitelist, the belt can simply hold anything a pocket can hold - Can easily be changed if it somehow becomes an issue
```

Comment:
This makes it as good as a toolbelt, not much of an issue honestly, but note that you can craft it so people may be making a lot of these as a substitute.

Reply:
Just a note, you can craft toolbelts too, which are a bit easier to craft. (It's just 4 leather in-hand, rather than a tablecraft recipe.)

Reply:
It also REQUIRES a toolbelt to craft it in the first place, and has one less storage slot, so the point of crafting it is kinda moot yeh